File: exercism/java/bob/src/main/java/Bob.java
public class Bob {


    public String hey(String statement) {
        if (statement.trim().isEmpty()) {
            return "Fine. Be that way!";
        } else if (capital(statement) && statement.trim().endsWith("?")) {
            return "Calm down, I know what I'm doing!";
        } else if (capital(statement)) {
            return "Whoa, chill out!";
        } else if (statement.trim().endsWith("?")) {
            return "Sure.";
        } else {
            return "Whatever.";
        }
    }


    public static boolean capital(String statement) {
        return statement.chars().filter(Character::isLetter).allMatch(Character::isUpperCase)
                && statement.chars().anyMatch(Character::isLetter);
    }
}

        

          

File: exercism/julia/bob/bob.jl
function bob(statement)
    if isblank(statement)
      "Fine. Be that way!"
    elseif capital(statement) && endswith(strip(statement), "!")
      "Whoa, chill out!"
    elseif capital(statement) && endswith(strip(statement), "?")
      "Calm down, I know what I'm doing!" 
    elseif capital(statement)
      "Whoa, chill out!"
    elseif endswith(strip(statement), "?")
      "Sure."
    else
      "Whatever."
    
    end
end

function isblank(input::AbstractString)
    length(input) == 0 || all(isspace, input)
end

function capital(statement::AbstractString)
    filter(isletter, statement) |> x -> all(isuppercase, x) && any(isletter, statement)
end
